### PR TITLE
Feat/pdf file upload and extraction edge function example

### PR DESCRIPTION
**Pull request by GeekyADAMS**
## What kind of change does this PR introduce?
Docs update:

Supabase > Docs > Guides > Functions
feature: pdf file upload and content extraction edge function example

## What is the current behavior?
Example use case does not currently exist

## What is the new behavior?
Edge Function docs example now has a guide for uploading pdf files to Supabase storage and extracting the file content.

### CODE REVIEW

**Review by ivasilov — DISMISSED** [review]

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts [hunk 1]
```diff
@@ -946,6 +946,10 @@ export const functions: NavMenuConstant = {
           name: 'Slack Bot responding to mentions',
           url: '/guides/functions/examples/slack-bot-mention',
         },
+        {
+          name: 'Upload PDF and extract content',
```
Comment (thorwebdev) [inline apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:950]:
```suggestion
          name: 'Convert PDF to HTML',
```

File: apps/docs/pages/guides/functions/examples/pdf-upload-extract.mdx [hunk 2]
```diff
@@ -0,0 +1,177 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  id: 'pdf-upload-extract',
+  title: 'Upload PDF and extract data',
+  description: 'Building function to upload a PDF file and extract data.',
+}
+
+## Upload pdf file to storage and extract its pure text content
```
Comment (thorwebdev) [inline apps/docs/pages/guides/functions/examples/pdf-upload-extract.mdx:9]:
```suggestion
## Upload a PDF file to storage and extract its content
```

File: examples/edge-functions/supabase/functions/upload-pdf-file-and-extract-content/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Upload pdf file to storage and extract its pure text content
```
Comment (thorwebdev) [inline examples/edge-functions/supabase/functions/upload-pdf-file-and-extract-content/README.md:1]:
```suggestion
# Upload a PDF file to storage and extract its content
```

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks 💚

**Comment by Sidebridge** [comment]
Hello @shivamsaboo17 ,
I found a better method to read pdf file content without using API2PDF API.

Implementation:

```
import pdf from 'npm:pdf-parse/lib/pdf-parse.js'

const fileBuffer = await getFileBuffer(file)

 if (!cvFile || !fileBuffer) {
    throw new UserError("Can'/t read pdf file")
 }

 const data = await pdf(fileBuffer)

 if (data) fileContent = data.text
 else throw new UserError('Error parsing PDF')
```

```
export const getFileBuffer = (uploadedFile: File): Promise<Buffer> => {
  const fileReader = new FileReader()

  // Create a promise for reading the file data
  const promise = new Promise<Buffer>((resolve, reject) => {
    fileReader.onload = () => {
      if (fileReader.result instanceof ArrayBuffer) {
        // const uint8Array = new Uint8Array(fileReader.result)

        resolve(fileReader.result as unknown as Buffer)
      } else {
        reject(new Error('Unable read uploaded PDF file'))
      }
    }

    fileReader.onerror = () => {
      reject(new Error('Error reading PDF file'))
    }
  })

  fileReader.readAsArrayBuffer(uploadedFile)

  // Wait for the file data to be read and return it
  return promise
}
```


I hope this helps

**Comment by GeekyADAMS** [comment]
Hello @shivamsaboo17 ,
I found a better method to read pdf file content without using API2PDF API.

Implementation:

```
import pdf from 'npm:pdf-parse/lib/pdf-parse.js'

const fileBuffer = await getFileBuffer(file)

 if (!cvFile || !fileBuffer) {
    throw new UserError("Can'/t read pdf file")
 }

 const data = await pdf(fileBuffer)

 if (data) fileContent = data.text
 else throw new UserError('Error parsing PDF')
```

```
export const getFileBuffer = (uploadedFile: File): Promise<Buffer> => {
  const fileReader = new FileReader()

  // Create a promise for reading the file data
  const promise = new Promise<Buffer>((resolve, reject) => {
    fileReader.onload = () => {
      if (fileReader.result instanceof ArrayBuffer) {
        // const uint8Array = new Uint8Array(fileReader.result)

        resolve(fileReader.result as unknown as Buffer)
      } else {
        reject(new Error('Unable read uploaded PDF file'))
      }
    }

    fileReader.onerror = () => {
      reject(new Error('Error reading PDF file'))
    }
  })

  fileReader.readAsArrayBuffer(uploadedFile)

  // Wait for the file data to be read and return it
  return promise
}
```


I hope this helps

**Pull request merged**